spec/services/streaming_service_provider_data_service_spec.rb

```diff
@@ -20,43 +20,36 @@
       it 'returns our default providers with a pay_model of "try"' do
         allow(Tmdb::Client).to receive(:request).and_return(nil)
 
-        results = StreamingServiceProviderDataService.get_providers(**movie_args)
-        aggregate_failures "all default providers appear with 'try'" do
-          expect(results.find {|r| r[:name] == 'Netflix'}[:pay_model] ).to eq('try')
-          expect(results.find {|r| r[:name] == 'Amazon Prime Video'}[:pay_model] ).to eq('try')
-          expect(results.find {|r| r[:name] == 'Amazon Video'}[:pay_model] ).to eq('try')
-          expect(results.find {|r| r[:name] == 'YouTube'}[:pay_model] ).to eq('try')
-          expect(results.find {|r| r[:name] == 'Vudu'}[:pay_model] ).to eq('try')
-        end
+        providers = StreamingServiceProviderDataService.get_providers(**movie_args)
+        try_pay_models = providers.select {|provider| provider[:pay_model] == 'try'}
+        expect(providers.length).to eq(try_pay_models.length)
       end
     end
 
     describe 'when the API does return results' do
+      let(:netflix_api_data) { {logo_path: "/5OAb2w.jpg", provider_id: 8, provider_name: "Netflix", display_priority: 26} }
+      let(:amazon_api_data) { {logo_path: "/5OAb2w.jpg", provider_id: 10, provider_name: "Amazon Video", display_priority: 26} }
+      let(:amazon_prime_api_data) { {logo_path: "/5OAb2w.jpg", provider_id: 9, provider_name: "Amazon Prime Video", display_priority: 25} }
+      let(:youtube_api_data) { {logo_path: "/xL9SUR.jpg", provider_id: 192, provider_name: "YouTube", display_priority: 51} }
+      let(:fandango_api_data) { {logo_path: "/peURlL.jpg", provider_id: 7, provider_name: "Fandango At Home", display_priority: 4} }
+
```

Comment:
These new `let`s clean up a lot of redundant data hashes throughout the spec.

Reply:
noice!